views/js/test/runner/proxy/qtiServiceProxy/test.js

```diff
@@ -0,0 +1,902 @@
+    'lodash',
+    'helpers',
+    'taoTests/runner/proxy',
+    'taoQtiTest/runner/proxy/qtiServiceProxy'
+], function($, _, helpers, proxyFactory, qtiServiceProxy) {
+    'use strict';
+
+    QUnit.module('qtiServiceProxy');
+
+
+    // backup/restore ajax method between each test
+    var ajaxBackup;
+    QUnit.testStart(function() {
+        ajaxBackup = $.ajax;
+    });
+    QUnit.testDone(function() {
+        $.ajax = ajaxBackup;
+    });
+
+
+    /**
+     * A simple AJAX mock factory that fakes a successful ajax call.
+     * To use it, just replace $.ajax with the returned value:
+     * <pre>$.ajax = ajaxMockSuccess(mockData);</pre>
+     * @param {*} response - The mock data used as response
+     * @param {Function} [validator] - An optional function called instead of the ajax method
+     * @returns {Function}
+     */
+    function ajaxMockSuccess(response, validator) {
+        var deferred = $.Deferred().resolve(response);
+        return function() {
+            validator && validator.apply(this, arguments);
+            return deferred.promise();
+        };
+    }
+
+
+    /**
+     * A simple AJAX mock factory that fakes a failing ajax call.
+     * To use it, just replace $.ajax with the returned value:
+     * <pre>$.ajax = ajaxMockError(mockData);</pre>
+     * @param {*} response - The mock data used as response
+     * @param {Function} [validator] - An optional function called instead of the ajax method
+     * @returns {Function}
+     */
+    function ajaxMockError(response, validator) {
+        var deferred = $.Deferred().reject(response);
+        return function() {
+            validator && validator.apply(this, arguments);
+            return deferred.promise();
+        };
+    }
+
+
+    QUnit.test('module', 6, function(assert) {
+        assert.equal(typeof qtiServiceProxy, 'object', "The qtiServiceProxy module exposes an object");
+        assert.equal(typeof proxyFactory, 'function', "The proxyFactory module exposes a function");
+        assert.equal(typeof proxyFactory.registerProxy, 'function', "The proxyFactory module exposes a registerProxy method");
+        assert.equal(typeof proxyFactory.getProxy, 'function', "The proxyFactory module exposes a getProxy method");
+
+        proxyFactory.registerProxy('qtiServiceProxy', qtiServiceProxy);
+
+        assert.equal(typeof proxyFactory('qtiServiceProxy'), 'object', "The proxyFactory factory has registered the qtiServiceProxy definition and produces an instance");
+        assert.notStrictEqual(proxyFactory('qtiServiceProxy'), proxyFactory('qtiServiceProxy'), "The proxyFactory factory provides a different instance of qtiServiceProxy on each call");
+    });
+
+
+    // small coverage check to facilitate dev of unit tests
+    var coverage = {};
+    QUnit.moduleDone(function() {
+        _.forEach(proxyApi, function(api) {
+            if (!coverage[api.name]) {
+                console.log('Missing unit test for method qtiServiceProxy.' + api.name);
```

Comment:
Blanket is already present, but has no effect.
This code is just here to have a quick list of remaining tests to implement regarding the declared api.